### PR TITLE
fix: move Networking Service to separate file with protocol

### PR DESCRIPTION
**Pull request by annabauza**
### Description

This is very simple example how we can create View Model with Service Protocol so we can use mocked service in tests

### Type of Change:
**Delete irrelevant options.**

- Code

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce. List any relevant details for your test.

### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: mentorship ios/Service/Networking/HomeApiService.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+//  Created for AnitaB.org Mentorship-iOS 
+//
+
+import Foundation
+import Combine
+
+class HomeApiService : HomeServiceProtocol {
+    
+    var homeResponseData : HomeModel.HomeResponseData?
+    var profileViewModel = ProfileViewModel()
```
Comment (ApheleiaS) [inline mentorship ios/Service/Networking/HomeApiService.swift:13]: Why do we have a reference to the ViewModel in the Services?

Comment (annabauza) [reply mentorship ios/Service/Networking/HomeApiService.swift:13]: Cause its a network model. I agree network model should not be in view model.

**Comment by vatsalkul** [comment]
@annabauza  Can you please update this PR, @yugantarjain will need to take this PR as a reference for refactoring the code.

File: mentorship ios/ViewModels/HomeViewModel.swift [hunk 2]
```diff
@@ -29,21 +30,17 @@ class HomeViewModel: ObservableObject {
...
-            .receive(on: RunLoop.main)
-            .catch { _ in Just(self.homeResponseData) }
-            .combineLatest(
-                NetworkManager.callAPI(urlString: URLStringConstants.Users.profile, token: token)
-                    .receive(on: RunLoop.main)
-                    .catch { _ in Just(self.profileViewModel.getProfile()) }
-            )
-            .sink { home, profile in
+        
+        service = homeService
```
Comment (annabauza) [inline mentorship ios/ViewModels/HomeViewModel.swift:34]: this should be in the view.

**Comment by sunjunkie** [comment]
@annabauza Can we close this PR?

**Comment by annabauza** [comment]
Yes please.

**Comment by sunjunkie** [comment]
Closed. See PR #102.

**Pull request closed without merging**